VisualRust/RustCommentSelectionCommandHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace VisualRust
+{
+    class RustCommentSelectionCommandHandler : VSCommandTarget<VSConstants.VSStd2KCmdID>
+    {
+        readonly ITextBuffer Buffer;
+        public RustCommentSelectionCommandHandler(IVsTextView vsTextView, IWpfTextView textView)
+            : base(vsTextView, textView)
+        {
+            Buffer = textView.TextBuffer;
+        }
+
+        protected override bool Execute(VSConstants.VSStd2KCmdID command, uint options, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            var snapshot = Buffer.CurrentSnapshot;
+            int start = TextView.Selection.Start.Position.Position;
+            int end = TextView.Selection.End.Position.Position;
+
+            // this is what we will store start offset in to get maximal indentation amount
+            int? insertStartOffset = null;
+
+            using (var edit = Buffer.CreateEdit())
+            {
+                while (start <= end)
+                {
+                    var line = snapshot.GetLineFromPosition(start);
+                    var text = line.GetText();
```

Comment:
Try to avoid getting all the text if you end up just checking its empty, use the Length property.
[Some official Microsoft word on it](http://blogs.msdn.com/b/noahric/archive/2010/07/13/q-amp-a-itextsnapshot-gettext.aspx)